### ISSUE TITLE
`azurerm_stream_analytics_stream_input_eventhub` throws error on `terraform import`

### ISSUE THREAD

**Issue by ThreeFive-O** [hashicorp/terraform-provider-azurerm#19039]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.1

### AzureRM Provider Version

3.29

### Affected Resource(s)/Data Source(s)

azurerm_stream_analytics_stream_input_eventhub

### Terraform Configuration Files

```hcl
resource "azurerm_stream_analytics_stream_input_eventhub" "evh" {
  name = "asa-evh-input"

  stream_analytics_job_name = "asa-job-example"
  resource_group_name       = "rg-eventhub"

  eventhub_consumer_group_name = "test"
  eventhub_name                = "evh-example"
  servicebus_namespace         = "evhns-example"
  authentication_mode          = "Msi"

  serialization {
    type     = "Json"
    encoding = "UTF8"
  }
}
```

### Debug Output/Panic Output

```shell
│ Error: converting Stream Input EventHub to an EventHub Stream Input: <nil>

TRACE output: (Refresh state)
{
  "id": "REDACTED",
  "name": "asa-evh-input",
  "type": "Microsoft.StreamAnalytics/streamingjobs/inputs",
  "properties": {
    "type": "Stream",
    "datasource": {
      "type": "Microsoft.EventHub/EventHub",
      "properties": {
        "consumerGroupName": "test",
        "eventHubName": "evh-example",
        "serviceBusNamespace": "evhns-example",
        "sharedAccessPolicyName": null,
        "authenticationMode": "Msi"
      }
    },
    "compression": {
      "type": "None"
    },
    "partitionKey": "partitionKeyExample",
    "serialization": {
      "type": "Json",
      "properties": {
        "encoding": "UTF8"
      }
    }
  }
}: timestamp=2022-10-28T12:00:53.597+0200
2022-10-28T12:00:53.598+0200 [ERROR] vertex "import azurerm_stream_analytics_stream_input_eventhub.evh result" error: converting Stream Input EventHub to an EventHub Stream Input: <nil>
```

### Expected Behaviour

`terraform import` succeeds

### Actual Behaviour

`terraform import` fails

### Steps to Reproduce

`terraform import azurerm_stream_analytics_stream_input_eventhub.evh <REDACTED ressource id>`

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. After tested, I can import the resource successfully. Could you double confirm whether the resource you intend to import is event hub type? Thanks.

![image](https://user-images.githubusercontent.com/19754191/198938680-b742c276-8e39-43da-a70f-e5391db265cb.png)

**Comment by ThreeFive-O** [comment]
Hi @neil-yechenwei, thank you for analyzing the issue.
As you can see from the debug output of the issue the type is Event Hub. This is the actual state that is loaded and produces an error when importing. 
```
    "datasource": {
      "type": "Microsoft.EventHub/EventHub",
```

**Comment by neil-yechenwei** [comment]
I assume the type you mentioned is parsed by Terraform. Suggest check the type in Azure Portal.

**Comment by ThreeFive-O** [comment]
Hi @neil-yechenwei, it is definitely an Event Hub input for the stream analytics job in Azure portal.

I've investigated further and I think I found the problem:
1. The Event Hub input I try to import was created in the Azure portal and already exists. The type the portal creates is ` "type": "Microsoft.EventHub/EventHub"`
2. If I create the Event Hub input with a terraform configuration, then delete the terraform state and then reimport the provisioned resource into the original config I get the following type `"type": "Microsoft.ServiceBus/EventHub"`

Terraform provisions with `Microsoft.ServiceBus` while Azure portal creates the input with `Microsoft.EventHub`.

Both show as Type "Event Hub" in the Azure portal like you mentioned in the screenshot.

The error cause is https://github.com/hashicorp/terraform-provider-azurerm/blob/04aa7716f96445577f53ef978cd40615c4a80412/internal/services/streamanalytics/stream_analytics_stream_input_eventhub_resource.go#L206

I substituted the conversion function `v.Datasource.AsEventHubStreamInputDataSource()` with `v.Datasource.AsEventHubV2StreamInputDataSource()` and recompiled the azurerm provider.

I'm now able to successfully import both the terraform provisioned event hub input `Microsoft.ServiceBus/EventHub` as well as the Azure portal provisioned event hub input `Microsoft.EventHub/EventHub`.

I'm not sure if this comes with any other downsides, but it likely seems to be the issue with the `Azure/azure-sdk-for-go`.

I would suggest to switch to `hashicorp/go-azure-sdk` anyways since this seems to be the general roadmap. Unfortunately I don't really have experience in doing so.

**Comment by ThreeFive-O** [comment]
@tombuildsstuff From what I can tell, the EventHubInputStream V1 and V2 have exactly the same payload. This can also be seen in the newer  `hashicorp/go-azure-sdk`.

The only difference is the type `Microsoft.ServiceBus/EventHub` (V1) and `Microsoft.EventHub/EventHub` V2.
I guess Microsoft just did an internal renaming of the data type and both point to the same thing.

With my Fix in #19068 the `Azure/azure-sdk-for-go` seems to parse the V1 and V2 version correctly for read, which is main problem of this ticket.

For Create (should be a new issue) maybe V2 should be the preferred variant as the Azure portal does the same thing.

As for Update V1 or V2 should be used dependent on the already provisioned version. I guess there is further work required after PR #19068 is merged.
